src/mmw/js/src/modeling/views.js

```diff
@@ -102,15 +106,58 @@ var ScenariosView = Marionette.LayoutView.extend({
 
     addScenario: function() {
         var scenario = new models.ScenarioModel({
-                name: 'New Scenario'
-            });
+            name: this.makeNewScenarioName()
+        });
 
         this.collection.add(scenario);
     },
 
+    counter: 0,
+
+    scenarioCounter: function() {
+        this.counter++;
+        if (this.counter < this.collection.length) {
+            this.counter = this.collection.length;
+        }
+        return this.counter;
+    },
+
+    /**
+     * Return a new scenario name of the format "New Scenario X" where X is a
+     * positive number and that is greater than all previous X.
+     */
+    makeNewScenarioName: function() {
+        // When making new scenarios, we need to make sure we don't
+        // accidentally give two the same name. Use a counter but ensure slug
+        // name is not in use by looking at all the current names.
+        var numbers = _.without(_.map(this.collection.models, function(model) {
+            var slug = model.get('slug')
+            var regEx = /^new-scenario-(\d)+/g;
+            if (slug.match(regEx) !== null && slug.match(regEx).length === 1) {
+                return parseInt(model.get('slug').replace(/^new-scenario-/g, ''));
+            }
+        }), undefined);
+        if (!_.isEmpty(numbers)) {
+            var max = _.max(numbers) + 1;
+            return 'New Scenario ' + max;
+        } else {
+            return 'New Scenario ' + this.scenarioCounter();
+        }
+    },
+
     setActiveScenario: function(e) {
-        var scenarioSlug = $(e.target).data('scenario-slug');
+        var scenarioSlug = $(e.currentTarget).data('scenario-slug');
 
+        _.each(this.collection.models, function(model) {
+            model.set('active', false);
+        }, this);
+
+        _.find(this.collection.models, function(model) {
+            return model.get('slug') === scenarioSlug;
+        }, this).set('active', true);
+
+        // we are trying to active a different tab so need to switch tabs here.
+        this.panelsRegion.$el.find('a[data-scenario-slug=' + scenarioSlug + ']').tab('show');
```

Comment:
This doesn't work without actually triggering it manually.

I'm going to tie the active_scnario_slug to the model's active property but we need the model for a scenario to know if it is active to it can add appropriate classes at render time.